pandas/tests/scalar/test_timestamp.py

```diff
@@ -1494,3 +1456,61 @@ def test_to_datetime_bijective(self):
         with tm.assert_produces_warning(exp_warning, check_stacklevel=False):
             assert (Timestamp(Timestamp.min.to_pydatetime()).value / 1000 ==
                     Timestamp.min.value / 1000)
+
+
+class TestTimestampEquivDateRange(object):
+    # Older tests in TestTimeSeries constructed their `stamp` objects
+    # using `date_range` instead of the `Timestamp` constructor.
+    # TestTimestampEquivDateRange checks that these are equivalent in the
+    # pertinent cases.
+
+    def test_date_range_timestamp_equiv(self):
+        rng = date_range('20090415', '20090519', tz='US/Eastern')
+        stamp = rng[0]
```

Comment:
Given that you use this construct several times across tests, you might want to consider making that a method.

Reply:
Given the large number of irons in the fire at the moment, would you be OK with these comments going into #17652 and being addressed in a follow-up?

Reply:
@jbrockmendel : I would prefer to make a push to refactor now, as this is a reorganization PR after all.  Save it in the end as a last commit to this PR.

Reply:
> I would prefer to make a push to refactor now, as this is a reorganization PR after all. Save it in the end as a last commit to this PR.

Alright, let's wrap this thing up!  Thanks for taking the time to review.

> Given that you use this construct several times across tests (i.e. rng = date_range(...)[0]), you might want to consider making that a separate method.

Seems like replacing a 1-line function call with a 1-line method call doesn't really gain anything, no?